addons/mass_mailing_sms/models/mailing_mailing.py

```diff
@@ -242,23 +266,40 @@ def action_send_mail(self, res_ids=None):
 
     def action_send_sms(self, res_ids=None):
         for mailing in self:
+            mailing_values = {
+                'sent_date': fields.Datetime.now(),
+                'kpi_mail_required': not mailing.sent_date,
+            }
             if not res_ids:
                 res_ids = mailing._get_remaining_recipients()
             if res_ids:
                 composer = self.env['sms.composer'].with_context(active_id=False).create(mailing._send_sms_get_composer_values(res_ids))
+                if not composer.mass_force_send:
+                    mailing_values['state'] = 'sending'
                 composer._action_send_sms()
-
-            mailing.write({
-                'state': 'done',
-                'sent_date': fields.Datetime.now(),
-                'kpi_mail_required': not mailing.sent_date,
-                })
+            mailing.write(mailing_values)
         return True
 
+    def _should_set_done_early(self):
```

Comment:
Meh. I don't like adding another method + override + sub-call just for that... will get back to it but would like to remove this (and/or find a way to avoid too much ping-pong).